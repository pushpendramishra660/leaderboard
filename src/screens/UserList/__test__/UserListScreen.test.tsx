import React from 'react';
import renderer from 'react-test-renderer';
import {Alert} from 'react-native';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import UserListScreen from '../index';

jest.mock('react-native/Libraries/Alert/Alert', () => ({
  alert: jest.fn(),
}));

describe('<UserListScreen />', () => {
  test('renders UserListScreen component', () => {
    const {getByText, getByPlaceholderText} = render(<UserListScreen />);
    const searchBarPlaceholder = getByPlaceholderText('User Name...');
    const searchButton = getByText('Search');
    expect(searchBarPlaceholder).toBeTruthy();
    expect(searchButton).toBeTruthy();
  });

  test('searching for a user displays user data', async () => {
    const {getByPlaceholderText, getByText} = render(<UserListScreen />);
    const searchBar = getByPlaceholderText('User Name...');
    const searchButton = getByText('Search');

    fireEvent.changeText(searchBar, 'Chris Buckley');
    fireEvent.press(searchButton);

    await waitFor(() => {
      expect(getByText('Chris Buckley')).toBeTruthy();
      expect(getByText('6200')).toBeTruthy();
    });
  });

  test('searching for a non-existing user shows error alert', () => {
    const {getByPlaceholderText, getByText} = render(<UserListScreen />);
    const searchBar = getByPlaceholderText('User Name...');
    const searchButton = getByText('Search');

    fireEvent.changeText(searchBar, 'NonExistingUser');
    fireEvent.press(searchButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      'Error',
      'User not found. Please enter a valid username.',
    );
  });

  it('UserListScreen matches snapshot', () => {
    const component = renderer.create(<UserListScreen />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
