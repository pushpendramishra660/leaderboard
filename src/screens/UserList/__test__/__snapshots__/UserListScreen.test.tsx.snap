// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListScreen /> UserListScreen matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "marginTop": 50,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F0F0F0",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/search.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginRight": 10,
            "width": 20,
          }
        }
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="User Name..."
        placeholderTextColor="#666D87BF"
        style={
          Object {
            "fontSize": 16,
            "width": "60%",
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#007bff",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "center",
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "center",
          }
        }
      >
        Rank
      </Text>
      <Text
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "center",
          }
        }
      >
        Number of Bananas
      </Text>
      <Text
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "center",
          }
        }
      >
        Is Searched User
      </Text>
    </View>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
